### PR TITLE
Added Mergesort in Js

### PR DESCRIPTION
**Pull request by adithyanarayana**
Please check all that apply.

#### This PR is for:
- [X] Algorithm
- [ ] Cheatsheet
- [ ] Resource
- [ ] Other

#### Programming Language: [Language here]

- [X] Have you implemented it on your local machine or any online compiler?
- [ ] There is the main( ) function to run the program and all the inputs which are being taken are clearly printed on the console or well commented in the file.
- [X] Filename is self-explanatory or there is Readme.md file explaining the necessary details of the algorithm.

### CODE REVIEW

File: JavaScript/Heapsort/Heapsort.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        heap_root(input, i);
+      }
+
+    for (i = input.length - 1; i > 0; i--) {
+        swap(input, 0, i);
+        array_length--;
+      
+      
+        heap_root(input, 0);
+    }
```
Comment (starkblaze01) [inline JavaScript/Heapsort/Heapsort.md:44]: Add return input after this.

**Review by starkblaze01 — CHANGES_REQUESTED** [review]

**Comment by adithyanarayana** [comment]
Hey @starkblaze01 Completed the necessary changes. Thank you

File: JavaScript/MergeSort/Mergesort.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+function merge_sort(left_part,right_part)
```
Comment (starkblaze01) [inline JavaScript/MergeSort/Mergesort.js:1]: Hi, @adithyanarayana almost there. Just a last few changes.
`merge_sort` --> `merge`, as it is called by this name in mergeSort funcion.

**Review by starkblaze01 — CHANGES_REQUESTED** [review]

File: JavaScript/MergeSort/Mergesort.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	}
+	return results;
+}
+
+function mergeSort(arr) {
+    if(arr.length <= 1) return arr;
+    var mid = Math.floor(arr.length/2);
+    var left = mergeSort(arr.slice(0,mid));
+    var right = mergeSort(arr.slice(mid));
+    return merge(left_part, right_part);
```
[the remark this answers is not in the document]
Comment (adithyanarayana) [reply JavaScript/MergeSort/Mergesort.js]: Done! completed

**Review by starkblaze01 — APPROVED** [review]
Thank you for your contribution 😄

**Pull request merged**